front/src/modules/rollingStock/components/RollingStockSelector/RollingStockSelector.tsx

```diff
@@ -1,35 +1,56 @@
-import React, { useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import RollingStockModal from 'modules/rollingStock/components/RollingStockSelector/RollingStockModal';
+import RollingStock2Img from 'modules/rollingStock/components/RollingStock2Img';
 import icon from 'assets/pictures/components/train.svg';
 import {
   comfort2pictogram,
   RollingStockInfo,
 } from 'modules/rollingStock/components/RollingStockSelector/RollingStockHelpers';
 import { useModal } from 'common/BootstrapSNCF/ModalSNCF';
-import { RollingStockComfortType, RollingStock } from 'common/api/osrdEditoastApi';
+import { RollingStockWithLiveries, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useTranslation } from 'react-i18next';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+import { useSelector } from 'react-redux';
 
 type RollingStockProps = {
-  rollingStockSelected?: RollingStock;
-  rollingStockComfort?: RollingStockComfortType;
-  image?: JSX.Element;
-  comfort?: string;
-  comfortType?: string;
-  choice?: string;
   condensed?: boolean;
 };
 
-const RollingStockSelector = ({
-  rollingStockSelected,
-  rollingStockComfort,
-  image,
-  comfort,
-  comfortType,
-  choice,
-  condensed,
-}: RollingStockProps) => {
+const RollingStockSelector = ({ condensed }: RollingStockProps) => {
   const { openModal } = useModal();
 
   const ref2scroll = useRef<HTMLDivElement>(null);
+
+  const [rollingStockSelected, setRollingStockSelected] = useState<
+    RollingStockWithLiveries | undefined
+  >();
+
+  const { t } = useTranslation(['translation', 'rollingstock']);
+  const { getRollingStockID, getRollingStockComfort } = useOsrdConfSelectors();
+  const rollingStockId = useSelector(getRollingStockID);
+  const rollingStockComfort = useSelector(getRollingStockComfort);
+  const comfort = t('rollingstock:comfort');
+  const comfortType = t(`rollingstock:comfortTypes.${rollingStockComfort}`);
+  const choice = t('rollingstock:rollingstockChoice');
```

Comment:
You can simplify the arguments and I think the consts aren't needed, you can use the `t()` directly in the jsx
```suggestion
  const comfort = t('comfort');
  const comfortType = t(`comfortTypes.${rollingStockComfort}`);
  const choice = t('rollingstockChoice');
```

Reply:
Also no need to use const, they will be re instantiated at each re-render. Directly put this in the html